corehq/apps/sms/views.py

```diff
@@ -140,10 +144,19 @@ def can_use_inbound_sms(self):
     def is_system_admin(self):
         return self.request.couch_user.is_superuser
 
+    @cached_property
+    def is_granted_messaging_access(self):
+        if settings.ENTERPRISE_MODE or self.domain_object.granted_messaging_access:
+            return True
+        subscription = Subscription.get_active_subscription_by_domain(self.domain_object)
+        if subscription is not None:
+            return subscription.plan_version.plan.edition == SoftwarePlanEdition.ENTERPRISE
```

Comment:
Ok, in that case let's do this.  We normally we would add an extra permission to the role for "granted messaging access" but this solution is fine:

```suggestion
            return subscription.plan_version.plan.name == DefaultProductPlan.get_default_plan_version(
                edition=SoftwarePlanEdition.ENTERPRISE
            ).plan.name
```

Reply:
Seems a little hacky to match on name, but I suppose it will work. There should be a more reliable way to do this check.